### ISSUE TITLE
Not a directory exception

### ISSUE THREAD

**Issue by tsteven4** [miurahr/aqtinstall#438]
Doing an install in a clean directory, and then repeating the install results in an exception on macos.

% aqt install-qt mac desktop 6.2.1 clang_64 -O /tmp/Qt
aqtinstall(aqt) v2.0.0 on Python 3.9.7 [CPython Clang 12.0.5 (clang-1205.0.22.9)]
Specified Qt version is unknown: 6.2.1.
Downloading qtbase...
Downloading qtdeclarative...
Downloading qtsvg...
Downloading qttools...
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Finished installation of qtsvg-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 2.65491444
Downloading qttranslations...
Redirected: qt.mirror.constant.com
Finished installation of qttranslations-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 3.05917851
Finished installation of qtbase-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 6.34495827
Finished installation of qttools-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 7.38393935
Finished installation of qtdeclarative-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 12.34958260
Patching /tmp/Qt/6.2.1/macos/bin/qmake
Finished installation
Time elapsed: 14.29340737 second
% aqt install-qt mac desktop 6.2.1 clang_64 -O /tmp/Qt
aqtinstall(aqt) v2.0.0 on Python 3.9.7 [CPython Clang 12.0.5 (clang-1205.0.22.9)]
Specified Qt version is unknown: 6.2.1.
Downloading qtbase...
Downloading qtdeclarative...
Downloading qtsvg...
Downloading qttools...
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Downloading qttranslations...
Redirected: qt.mirror.constant.com
Finished installation of qttranslations-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 3.06217552
[Errno 20] Not a directory: '/tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Headers'
multiprocessing.pool.RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 51, in starmapstar
    return list(itertools.starmap(args[0], args[1]))
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 893, in installer
    szf.extractall(path=base_dir)
  File "/usr/local/lib/python3.9/site-packages/py7zr/py7zr.py", line 948, in extractall
    self._extract(path=path, return_dict=False, callback=callback)
  File "/usr/local/lib/python3.9/site-packages/py7zr/py7zr.py", line 604, in _extract
    self.worker.extract(
  File "/usr/local/lib/python3.9/site-packages/py7zr/py7zr.py", line 1165, in extract
    self.extract_single(
  File "/usr/local/lib/python3.9/site-packages/py7zr/py7zr.py", line 1276, in extract_single
    raise e
  File "/usr/local/lib/python3.9/site-packages/py7zr/py7zr.py", line 1258, in extract_single
    with fileish.open(mode="wb") as ofp:
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/pathlib.py", line 1252, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/pathlib.py", line 1120, in _opener
    return self._accessor.open(self, flags, mode)
NotADirectoryError: [Errno 20] Not a directory: '/tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Headers'
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 99, in run
    args.func(args)
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 294, in run_install_qt
    run_installer(qt_archives.get_packages(), base_dir, sevenzip, keep)
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 835, in run_installer
    pool.starmap(installer, tasks)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 372, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
NotADirectoryError: [Errno 20] Not a directory: '/tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Headers'
Arguments: `['/usr/local/bin/aqt', 'install-qt', 'mac', 'desktop', '6.2.1', 'clang_64', '-O', '/tmp/Qt']` Host: `uname_result(system='Darwin', node='27229.local', release='20.6.0', version='Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64', machine='x86_64')`
===========================PLEASE FILE A BUG REPORT===========================
You have discovered a bug in aqt.
Please file a bug report at https://github.com/miurahr/aqtinstall/issues.
Please remember to include a copy of this program's output in your report.
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/threading.py", line 973, in _bootstrap_inner

**Comment by miurahr** [comment]
It seems a bug on py7zr library. Could you try `--external` flag to specify 7zip command line program?

The archive which produce error may have a symbolic  link,  '/tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Headers', and 
have a file entry under it.
py7zr make symbolic link after all the files are extracted, to avoid cyclic/infinite extraction looping.

**Comment by tsteven4** [comment]
Sure, here is the result.  The TypeError seems like a different issue.  I did update aqtinstall and all its dependencies.

+ /bin/rm -fr /tmp/Qt
+ aqt install-qt mac desktop 6.2.1 clang_64 -O /tmp/Qt -E /usr/local/bin/7z
aqtinstall(aqt) v2.0.2 on Python 3.9.7 [CPython Clang 13.0.0 (clang-1300.0.29.3)]
Downloading qtdeclarative...
Downloading qtbase...
Downloading qtsvg...
Downloading qttools...
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Finished installation of qtsvg-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 2.69209237
Downloading qttranslations...
Redirected: qt.mirror.constant.com
Finished installation of qttranslations-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 3.05180183
Finished installation of qtbase-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 5.84071561
Finished installation of qttools-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 6.82667982
Finished installation of qtdeclarative-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 10.05013753
Patching /tmp/Qt/6.2.1/macos/bin/qmake
Finished installation
Time elapsed: 11.92154962 second
+ aqt install-qt mac desktop 6.2.1 clang_64 -O /tmp/Qt -E /usr/local/bin/7z
aqtinstall(aqt) v2.0.2 on Python 3.9.7 [CPython Clang 13.0.0 (clang-1300.0.29.3)]
Downloading qtdeclarative...
Downloading qtbase...
Downloading qtsvg...
Downloading qttools...
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
Redirected: qt.mirror.constant.com
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSvgWidgets.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSvg.framework/QtSvg
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSvgWidgets.framework/QtSvgWidgets
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSvgWidgets.framework/Resources
Downloading qttranslations...
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtConcurrent.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtCore.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDBus.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtGui.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtNetwork.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtOpenGL.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtOpenGLWidgets.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtPrintSupport.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSql.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtTest.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtWidgets.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtXml.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtConcurrent.framework/QtConcurrent
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtCore.framework/QtCore
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDBus.framework/QtDBus
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDesigner.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDesignerComponents.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtHelp.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtUiTools.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtGui.framework/QtGui
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtNetwork.framework/QtNetwork
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtOpenGL.framework/QtOpenGL
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtOpenGLWidgets.framework/QtOpenGLWidgets
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtPrintSupport.framework/QtPrintSupport
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDesigner.framework/QtDesigner
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSql.framework/QtSql
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtTest.framework/QtTest
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDesignerComponents.framework/QtDesignerComponents
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtHelp.framework/QtHelp
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtUiTools.framework/QtUiTools
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDesigner.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDesignerComponents.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtHelp.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtUiTools.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtWidgets.framework/QtWidgets
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtXml.framework/QtXml
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtConcurrent.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtCore.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtDBus.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtGui.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtNetwork.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtOpenGL.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtOpenGLWidgets.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtPrintSupport.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtSql.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtTest.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtWidgets.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtXml.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsAnimation.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsFolderListModel.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsQmlModels.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsSettings.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsSharedImage.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsWavefrontMesh.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQml.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlCore.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlLocalStorage.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlModels.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlWorkerScript.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlXmlListModel.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuick.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickControls2.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickControls2Impl.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2QuickImpl.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2Utils.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickLayouts.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickParticles.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickShapes.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickTemplates2.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickTest.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickWidgets.framework/Headers
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsAnimation.framework/QtLabsAnimation
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsFolderListModel.framework/QtLabsFolderListModel
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsQmlModels.framework/QtLabsQmlModels
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsSettings.framework/QtLabsSettings
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsSharedImage.framework/QtLabsSharedImage
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsWavefrontMesh.framework/QtLabsWavefrontMesh
Redirected: qt.mirror.constant.com
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQml.framework/QtQml
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlCore.framework/QtQmlCore
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlLocalStorage.framework/QtQmlLocalStorage
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlModels.framework/QtQmlModels
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlWorkerScript.framework/QtQmlWorkerScript
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlXmlListModel.framework/QtQmlXmlListModel
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuick.framework/QtQuick
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickControls2.framework/QtQuickControls2
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickControls2Impl.framework/QtQuickControls2Impl
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2.framework/QtQuickDialogs2
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2QuickImpl.framework/QtQuickDialogs2QuickImpl
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2Utils.framework/QtQuickDialogs2Utils
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickLayouts.framework/QtQuickLayouts
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickParticles.framework/QtQuickParticles
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickShapes.framework/QtQuickShapes
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickTemplates2.framework/QtQuickTemplates2
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickTest.framework/QtQuickTest
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickWidgets.framework/QtQuickWidgets
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsAnimation.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsFolderListModel.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsQmlModels.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsSettings.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsSharedImage.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtLabsWavefrontMesh.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQml.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlCore.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlLocalStorage.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlModels.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlWorkerScript.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQmlXmlListModel.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuick.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickControls2.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickControls2Impl.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2QuickImpl.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickDialogs2Utils.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickLayouts.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickParticles.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickShapes.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickTemplates2.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickTest.framework/Resources
ERROR: Can not open output file : No such file or directory : /tmp/Qt/6.2.1/macos/lib/QtQuickWidgets.framework/Resources
Finished installation of qttranslations-MacOS-MacOS_11_00-Clang-MacOS-MacOS_11_00-X86_64-ARM64.7z in 3.10335253
Caught TypeError, terminating installer workers
sequence item 1: expected str instance, bytes found
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 99, in run
    args.func(args)
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 298, in run_install_qt
    run_installer(qt_archives.get_packages(), base_dir, sevenzip, keep)
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 883, in run_installer
    raise e from e
  File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 859, in run_installer
    pool.starmap(installer, tasks)
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 372, in starmap
    return self._map_async(func, iterable, starmapstar, chunksize).get()
  File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 771, in get
    raise self._value
TypeError: sequence item 1: expected str instance, bytes found
aqtinstall(aqt) v2.0.2 on Python 3.9.7 [CPython Clang 13.0.0 (clang-1300.0.29.3)]
Working dir: `/Users/administrator`
Arguments: `['/usr/local/bin/aqt', 'install-qt', 'mac', 'desktop', '6.2.1', 'clang_64', '-O', '/tmp/Qt', '-E', '/usr/local/bin/7z']` Host: `uname_result(system='Darwin', node='27229.local', release='20.6.0', version='Darwin Kernel Version 20.6.0: Tue Oct 12 18:33:42 PDT 2021; root:xnu-7195.141.8~1/RELEASE_X86_64', machine='x86_64')`
===========================PLEASE FILE A BUG REPORT===========================
You have discovered a bug in aqt.
Please file a bug report at https://github.com/miurahr/aqtinstall/issues.
Please remember to include a copy of this program's output in your report.

**Comment by miurahr** [comment]
> Doing an install in a clean directory, and then repeating the install results in an exception on macos.

It should be failed.

```
[Errno 20] Not a directory: '/tmp/Qt/6.2.1/macos/lib/QtSvg.framework/Headers'
```

Headers is symbolic link. Extractor try to make symbolic link as same file, but there has already been a link, so failed.

Normal file can overwrite, but OS do not allow make link on existent link file.

**Comment by ddalcino** [comment]
If this is a "won't fix", then I think we should catch the error in some way to prevent the "PLEASE FILE A BUG REPORT" message.

The "ERROR: Can not open output file" error should have been caught here:
https://github.com/miurahr/aqtinstall/blob/dcb217f74631c55e72d07ce504ad148fc9d83605/aqt/installer.py#L950-L955

I'm not clear on why `7z` didn't return nonzero here. The `7z` manpage says that it only returns 0 when no errors occur. If `7z` had returned nonzero, `subprocess.run()` would raise `CalledProcessError` because `check=True`. This did not happen. Is there something else we can do to check that the extraction operation succeeded?

>```
>File "/usr/local/lib/python3.9/site-packages/aqt/installer.py", line 859, in run_installer
>pool.starmap(installer, tasks)
>File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 372, in starmap
>return self._map_async(func, iterable, starmapstar, chunksize).get()
>File "/usr/local/Cellar/python@3.9/3.9.7_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/multiprocessing/pool.py", line 771, in get
>raise self._value
>TypeError: sequence item 1: expected str instance, bytes found
>```

What's going on here with the `TypeError`? I'm having trouble tracking the source of this. To me, it looks like `multiprocessing.pool.ApplyResult._value` was a TypeError, but it doesn't say where it came from. Are we passing a `bytes` object into a function that requires a `str` somewhere in `aqt.installer.installer()`?

**Comment by miurahr** [comment]
I've added a test case  to extract twice the file reported here on py7zr project.
https://github.com/miurahr/py7zr/pull/383

It reproduce the exception.  This seems a bug on py7zr project.

**Comment by miurahr** [comment]
py7zr released the fix as v0.17.0, and #462 bump py7zr@v0.17.0.
I hope #462 resolve here.

**Comment by tsteven4** [comment]
I tried aqt 2.0.2 with py7zr-0.16.2 and I could reproduce the failure as above.
I tried aqt 2.0.2 with py7zr-0.17.0, and I could NOT reproduce the failure.  It seems py7zr-0.17.0 resolves my issue.
Thanks
